syntax/parse.go

```diff
@@ -16,6 +16,9 @@ import "log"
 // Enable this flag to print the token stream and log.Fatal on the first error.
 const debug = false
 
+// Enable this flag to attach comments to the AST.
+const keepComments = false
```

Comment:
Naturally this should be a mode flag to the parser, not a compile-time constant.  Add:

```
  type Mode uint
  const (
     RetainComments Mode = 1<<iota
  )
  func Parse(..., mode Mode)
  func ParseExpr(, ...mode Mode)
```

This will be a breaking API change, but so be it.

Reply:
Done